<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Spatial Analysis of Gene Diversity — sHe • biotools</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Spatial Analysis of Gene Diversity — sHe" />
<meta property="og:description" content="Estimate spatial gene diversity (expected heterozygozity - He) through the individual-centred approach by Manel et al. (2007).
sHe() calculates the unbiased estimate of He based on the information of allele frequency obtained from codominant or
dominant markers in individuals within a circular moving windows of known radius over the sampling area." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biotools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/overview.html">biotools: an overview</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/arsilva87/biotools/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Spatial Analysis of Gene Diversity</h1>
    
    <div class="hidden name"><code>sHe.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Estimate spatial gene diversity (expected heterozygozity - <em>He</em>) through the individual-centred approach by Manel et al. (2007).
<code>sHe()</code> calculates the unbiased estimate of <em>He</em> based on the information of allele frequency obtained from codominant or
dominant markers in individuals within a circular moving windows of known radius over the sampling area.</p>
    </div>

    <pre class="usage"><span class='fu'>sHe</span><span class='op'>(</span><span class='va'>x</span>, coord.cols <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span>, marker.cols <span class='op'>=</span> <span class='fl'>3</span><span class='op'>:</span><span class='fl'>4</span>, 
  marker.type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"codominant"</span>, <span class='st'>"dominant"</span><span class='op'>)</span>, 
  grid <span class='op'>=</span> <span class='cn'>NULL</span>, latlong2km <span class='op'>=</span> <span class='cn'>TRUE</span>, <span class='va'>radius</span>, nmin <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>a data frame or numeric matrix containing columns with coordinates of individuals and marker genotyping</p></td>
    </tr>
    <tr>
      <th>coord.cols</th>
      <td><p>a vector of integer giving the columns of coordinates in <code>x</code></p></td>
    </tr>
    <tr>
      <th>marker.cols</th>
      <td><p>a vector of integer giving the columns of markers in <code>x</code></p></td>
    </tr>
    <tr>
      <th>marker.type</th>
      <td><p>a character; the type of molecular marker</p></td>
    </tr>
    <tr>
      <th>grid</th>
      <td><p>optional; a two-column matrix containing coordinates over which to predict <em>He</em></p></td>
    </tr>
    <tr>
      <th>latlong2km</th>
      <td><p>logical; should coordinates be converted from lat/long format into kilometer-grid based?</p></td>
    </tr>
    <tr>
      <th>radius</th>
      <td><p>the radius of the moving window. It must be in the same format as sampling coordinates</p></td>
    </tr>
    <tr>
      <th>nmin</th>
      <td><p>optional; a numeric value indicating the minimum number of individuals used to calculate <em>He</em>. If is
	the number of individuals in a certain location is less then <code>nmin</code>, <code>sHe</code> will consider <em>He</em> as zero.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The unbiased estimate of expected heterogygozity (Nei, 1978) is given by:</p>
<p>$$ He = (1 - \sum_{i=1}^{n} p_{i}^{2}) \frac{2n}{2n - 1} $$</p>
<p>where \(p_{i}\) is the frequency of the i-th allele per locus considering the \(n\) individuals in a certain location.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list of</p>
<dt>diversity</dt><dd><p>a data frame with the following columns: <em>coord.x</em> - the x-axis coordinates of the predicion grid,
	<em>coord.y</em> - the y-axis coordinates of the predicion grid, <em>n</em> - the number of individuals in a certain points in the grid,
	<em>MaxDist</em> - the maximum observed distance among these individuals, <em>uHe</em> - the unbiased estimate
	of gene diversity (as expressed above), and <em>SE</em> - the standard error of <em>uHe</em>.</p></dd>
  <dt>mHe</dt><dd><p>a matrix containing the estimates of <em>He</em> for every marker, on each point of the <code>grid</code>.</p></dd>
  <dt>locations</dt><dd><p>a numeric matrix containing the sampling coordinates, as provides as input.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>da Silva, A.R.; Malafaia, G.; Menezes, I.P.P. (2017) biotools: an R function to predict 
   spatial gene diversity via an individual-based approach. <em>Genetics and Molecular Research</em>, 
   <b>16</b>: gmr16029655.</p>
<p>Manel, S., Berthoud, F., Bellemain, E., Gaudeul, M., Luikart, G., Swenson, J.E., Waits, L.P., 
	Taberlet, P.; Intrabiodiv Consortium. (2007) A new individual-based spatial approach for 
	identifying genetic discontinuities in natural populations. <em>Molecular Ecology</em>, <b>16</b>:2031-2043.</p>
<p>Nei, M. (1978) Estimation of average heterozygozity and genetic distance from a small number of individuals. 
	<em>Genetics</em>, <b>89</b>: 583-590.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;</p>
<p>Ivandilson Pessoa Pinto de Menezes &lt;ivan.menezes@ifgoiano.edu.br&gt;</p>
    <h2 class="hasAnchor" id="warning-"><a class="anchor" href="#warning-"></a>Warning </h2>

    <p>Depending on the dimension of <code>x</code> and/or <code>grid</code>, <code>sHe()</code> can be time demanding.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>levelplot</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>moco</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>brazil</span><span class='op'>)</span>

<span class='co'># check points</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span><span class='va'>brazil</span>, cex <span class='op'>=</span> <span class='fl'>0.1</span>, col <span class='op'>=</span> <span class='st'>"gray"</span><span class='op'>)</span>
</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/graphics/points.html'>points</a></span><span class='op'>(</span><span class='va'>Lat</span> <span class='op'>~</span> <span class='va'>Lon</span>, data <span class='op'>=</span> <span class='va'>moco</span>, col <span class='op'>=</span> <span class='st'>"blue"</span>, pch <span class='op'>=</span> <span class='fl'>20</span><span class='op'>)</span>
</div><div class='img'><img src='sHe-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'># using a retangular grid (not passed as input!)</span>
<span class='co'># ex &lt;- sHe(x = moco, coord.cols = 1:2,</span>
<span class='co'>#  marker.cols = 3:20, marker.type = "codominant",</span>
<span class='co'>#  grid = NULL, radius = 150)</span>
<span class='co'>#ex</span>
<span class='co'># plot(ex, xlab = "Lon", ylab = "Lat")</span>

<span class='co'># A FANCIER PLOT...</span>
<span class='co'># using Brazil's coordinates as prediction grid</span>
<span class='co'># ex2 &lt;- sHe(x = moco, coord.cols = 1:2,</span>
<span class='co'>#  marker.cols = 3:20, marker.type = "codominant",</span>
<span class='co'>#  grid = brazil, radius = 150)</span>
<span class='co'># ex2</span>
<span class='co'>#</span>
<span class='co'># library(maps)</span>
<span class='co'># borders &lt;- data.frame(x = map("world", "brazil")$x, </span>
<span class='co'>#  y = map("world", "brazil")$y)</span>
<span class='co'>#</span>
<span class='co'># library(latticeExtra)</span>
<span class='co'># plot(ex2, xlab = "Lon", ylab = "Lat", </span>
<span class='co'>#  xlim = c(-75, -30), ylim = c(-35, 10), aspect = "iso") +</span>
<span class='co'>#   latticeExtra::as.layer(xyplot(y ~ x, data = borders, type = "l")) +</span>
<span class='co'>#   latticeExtra::as.layer(xyplot(Lat ~ Lon, data = moco))</span>

<span class='co'># End (not run)</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Anderson Rodrigo da Silva.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


